azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -181,28 +181,28 @@ def _init_impl(self):
         try:
             self.msi_auth_context = ManagedIdentityCredential(**self._msi_args)
         except Exception as e:
-            raise KustoClientError("Failed to initialize MSI ManagedIdentityCredential with [" + str(self._msi_params) + "]\n" + str(e))
+            raise KustoClientError("Failed to initialize MSI ManagedIdentityCredential with [" + str(self._msi_args) + "]\n" + str(e))
 
-    def _get_token_impl(self) -> dict:
+    def _get_token_impl(self) -> Optional[dict]:
         return None
 
     def _get_token_from_cache_impl(self) -> dict:
         try:
             msi_token = self.msi_auth_context.get_token(self._kusto_uri)
             return {TokenConstants.MSAL_TOKEN_TYPE: TokenConstants.BEARER_TYPE, TokenConstants.MSAL_ACCESS_TOKEN: msi_token.token}
         except Exception as e:
-            raise KustoClientError("Failed to obtain MSI token for '" + self._kusto_uri + "' with [" + str(self._msi_params) + "]\n" + str(e))
+            raise KustoClientError("Failed to obtain MSI token for '" + self._kusto_uri + "' with [" + str(self._msi_args) + "]\n" + str(e))
 
 
 class AzCliTokenProvider(TokenProviderBase):
     """ AzCli Token Provider obtains a refresh token from the AzCli cache and uses it to authenticate with MSAL """
 
     def __init__(self, kusto_uri: str):
         super().__init__(kusto_uri)
-        self._msal_client = None
-        self._client_id = None
+        self._msal_client = None  # type: Optional[ClientApplication]

```

Comment:
Assuming we drop 3.5, there's a syntax to type variables
Need to make sure it is supported in 3.6

Reply:
yes, the cool syntax was only added to 3.6:
https://www.python.org/dev/peps/pep-0526/

Notice that I did commit with that syntax, but reverted it after the 3.5 build failed.

Reply:
+ @ArielYehezkely 
Where do we stand on deprecating Python 3.5 support?

Reply:
I don't know, we just moved to version 2.0.0, maybe we can stop support it now despite ideally I would have removed it before bumping a major version.

Reply:
Ok I think that for this PR at least, we will leave in the support.

Do I need to update the package version, if so to what number, is this a minor or a minor minor?

Reply:
No need to update now, we usually aggregate few changes and publish a new release once in a while.